http-gateway/service/httpApi.go

```diff
@@ -6,120 +6,225 @@ import (
 	"net/http"
 	"net/http/httptest"
 	"net/http/httputil"
+	"net/url"
 	"strings"
 
-	pbCA "github.com/plgd-dev/cloud/certificate-authority/pb"
+	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
 	"github.com/plgd-dev/cloud/grpc-gateway/client"
+	"github.com/plgd-dev/cloud/grpc-gateway/pb"
 	"github.com/plgd-dev/cloud/http-gateway/uri"
 	"github.com/plgd-dev/cloud/pkg/log"
-	kitNetGrpc "github.com/plgd-dev/cloud/pkg/net/grpc"
 	kitHttp "github.com/plgd-dev/cloud/pkg/net/http"
-	raClient "github.com/plgd-dev/cloud/resource-aggregate/client"
+	"github.com/tmc/grpc-websocket-proxy/wsproxy"
 
-	"github.com/google/uuid"
 	router "github.com/gorilla/mux"
 )
 
 //RequestHandler for handling incoming request
 type RequestHandler struct {
-	client   *client.Client
-	caClient pbCA.CertificateAuthorityClient
-	config   *Config
-	manager  *ObservationManager
-	raClient *raClient.Client
+	client *client.Client
+	config *Config
+	mux    *runtime.ServeMux
 }
 
 //NewRequestHandler factory for new RequestHandler
-func NewRequestHandler(client *client.Client, caClient pbCA.CertificateAuthorityClient, config *Config, manager *ObservationManager, raClient *raClient.Client) *RequestHandler {
+func NewRequestHandler(config *Config, client *client.Client) *RequestHandler {
 	return &RequestHandler{
-		client:   client,
-		config:   config,
-		manager:  manager,
-		caClient: caClient,
-		raClient: raClient,
+		client: client,
+		config: config,
+		mux: runtime.NewServeMux(
+			runtime.WithErrorHandler(errorHandler),
+			runtime.WithMarshalerOption(uri.ApplicationProtoJsonContentType, newJsonpbMarshaler()),
+			runtime.WithMarshalerOption(runtime.MIMEWildcard, newJsonMarshaler()),
+		),
 	}
 }
 
-func resourceMatcher(r *http.Request, rm *router.RouteMatch) bool {
+func loggingMiddleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		data, err := httputil.DumpRequest(r, false)
+		if err != nil {
+			log.Infof("Request: %v %v", r.Method, r.RequestURI)
+		} else {
+			log.Infof("Request: %v", string(data))
+		}
+
+		// Call the next handler, which can be another middleware in the chain, or the final handler.
+		next.ServeHTTP(w, r)
+	})
+}
+
+func correlationIDHeaderToQuery(next http.Handler) http.Handler {
```

Comment:
Didn't we discuss that it's not configurable from the client, but automatically generated internally in the grpc gw?

Reply:
removed